### PR TITLE
Sanitize grouping values used as filenames and link hrefs

### PR DESCRIPTION
**Pull request by stevelandeydescript**
Grouping values (surface names, team names) were used raw as both filenames and link hrefs in HtmlWriter.toGroupingPages(). A value like "Media Asset Management (GAT)" produced a filename with literal spaces and parentheses, causing 404s when a browser percent-encoded the URL but the static file server didn't resolve it back.

Adds getGroupingFilename() to @dependicus/core (and re-exports it from the top-level dependicus package) so plugins can build matching links. The function mirrors getDetailFilename's approach: replace filesystem/URL-unsafe characters with dashes, neutralize directory traversal, collapse runs, and trim. Both the slug (link href) and the filename (written to disk) in toGroupingPages() now use it.

---

> [!NOTE]
> **Low Risk**
> Low risk: change is isolated to how grouping page filenames/hrefs are generated, plus adds a small `slugify` dependency and associated lockfile updates.
> 
> **Overview**
> Fixes grouping detail page 404s by **slugifying grouping values** (teams/surfaces) before using them as filenames and link hrefs.
> 
> Introduces `getGroupingFilename()` in `@dependicus/core` (and re-exports it from `dependicus`), adds tests for edge cases (spaces, punctuation, traversal, empty), and updates `HtmlWriter.toGroupingPages()` to use the helper consistently. Dependency manifests/lockfiles are updated to include `slugify` and bump the `dependicus` package version to `0.2.0-rc.2`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 12d96c7bed41d02412f5b3db44d5037791d9c6e5. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/core/src/types.ts [hunk 1]
```diff
@@ -211,6 +211,18 @@ export function getDetailFilename(name: string, version: string): string {
     return `${safeName}@${safeVersion}.html`;
 }
 
+/** Build the filename for a grouping detail page (e.g., "media-asset-management-gat.html"). */
+export function getGroupingFilename(value: string): string {
+    return (
+        value
+            .replace(/[/:"<>|*?()]/g, '-')
```
[the remark this answers is not in the document]
Comment (lorenzo-descript) [reply packages/core/src/types.ts]: Yeah need to sanitize more stuff for sure

**Review by lorenzo-descript — APPROVED** [review]

File: packages/core/src/types.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (stevelandeydescript) [reply packages/core/src/types.ts]: Yeah, brought in a dependency

File: packages/core/src/types.ts [hunk 2]
```diff
@@ -211,6 +212,12 @@ export function getDetailFilename(name: string, version: string): string {
     return `${safeName}@${safeVersion}.html`;
 }
 
+/** Build the filename for a grouping detail page (e.g., "Media-Asset-Management-GAT.html"). */
+export function getGroupingFilename(value: string): string {
+    const slug = slugify(value, { strict: true });
+    return `${slug || 'unknown'}.html`;
+}
```
[the remark this answers is not in the document]
Comment (stevelandeydescript) [reply packages/core/src/types.ts:219]: I think this is a reasonable tradeoff.

File: packages/core/package.json [hunk 3]
```diff
@@ -25,6 +25,7 @@
         "handlebars": "^4.7.8",
         "js-yaml": "^4.1.1",
         "semver": "^7.7.4",
+        "slugify": "^1.6.9",
```
[the remark this answers is not in the document]
Comment (stevelandeydescript) [reply packages/core/package.json:28]: This project doesn't use catalogs
